tests/acceptance/test_bootstrap_artifact.py

```diff
@@ -0,0 +1,141 @@
+            prepared_test_build_base["bitbake_corebase"],
+            bitbake_image,
+            [
+                'MENDER_ARTIFACT_SIGNING_KEY = "%s"'
+                % os.path.join(os.getcwd(), signing_key("RSA").private),
+                'MENDER_ARTIFACT_VERIFY_KEY = "%s"'
+                % os.path.join(os.getcwd(), signing_key("RSA").public),
+                'SYSTEMD_AUTO_ENABLE:pn-mender-client = "enable"',
+            ],
+        )
+        return prepared_test_build_base["build_dir"]
+
+    return img_builder
+
+
+@pytest.fixture(scope="function")
+def boot_device_with_bootstrap_image(
+    request,
+    qemu_wrapper,
+    session_connection,
+    board_type,
+    conversion,
+    build_image_with_signed_bootstrap_artifact,
+):
+
+    """
+
+    Simple override of the global setup_board fixture with 'function' local scope.
+
+    This means that the board is booted, and brought down as a part of the same
+    test function, as opposed to ran as a part of all the tests, as the global
+    test fixture does.
+
+    """
+
+    print("board type: ", board_type)
+
+    if "qemu" in board_type:
+        image_dir = build_image_with_signed_bootstrap_artifact()
+        return setup_qemu(request, qemu_wrapper, image_dir, session_connection)
+    elif "raspberrypi4" in board_type and request.config.getoption(
+        "--hardware-testing"
+    ):
+        return setup_hardware_test_board(request, session_connection)
+    elif conversion:
+        pytest.skip("Skip non-qemu platforms for mender-convert")
+    else:
+        pytest.fail("unsupported board type {}".format(board_type))
+
+    # Make sure 'image.dat' is not present on the device
+    session_connection.run("rm -f image.dat")
+
+
+@pytest.mark.min_mender_version("3.4.0")
+@pytest.mark.min_yocto_version("dunfell")
+@pytest.mark.only_with_image("ext4", "ext3", "ext2")
+def test_bootstrap_artifact_install(
+    request, boot_device_with_bootstrap_image, connection,
+):
+    """Test that the Mender Bootstrap Artifact works correctly
+
+    # 1. - Build an image with a signed bootstrap Artifact
+
+    # 2. - Boot a device with the given image
+
+    # 3. - Verify that the bootstrap artifact has populated the DB, and is deleted
+
+    # 4. Bring the device down
+
+    """
+
+    # Give the client a little bit of time to install the Artifact
+    for i in range(10):
+        if "bootstrap.mender" not in connection.run("ls /data/mender/").stdout.split():
+            break
+        time.sleep(10)
+
+    # Check for the presence of the bootstrap Artifact
+    # This should not be present, if the client has installed it
+    assert "bootstrap.mender" not in connection.run("ls /data/mender/").stdout.split()
+
+    # Check that the database of the client has been populated
+    device_provides = connection.run("mender show-provides").stdout.strip()
+    assert "rootfs-image.checksum" in device_provides
+    assert "rootfs-image.version" in device_provides
```

Comment:
I would also assert that `mender show-artifact` returns something different than `unknown`.